recipes/full_finetune_distributed.py

```diff
@@ -151,13 +151,20 @@ def setup(self, cfg: DictConfig) -> None:
 
         ckpt_dict = self.load_checkpoint(cfg.checkpointer)
 
+        # TODO: This is a temporary fix to handle the model tie for GEMMA models.
+        if cfg.checkpointer.model_type == "GEMMA":
+            model_tie = True
+        else:
+            model_tie = False
```

Comment:
We could also consider associating a weight tying config explicitly with the model type and using that in the checkpointer. E.g. 

```
@dataclass
ModelType
	name: str
	weight_tying_config: Dict[str, str] = field(default_factory=dict)
```

Then Gemma would be `ModelType(name="GEMMA", weight_tying_config={"tok_embeddings.weight": "output.weight"}`

(Anyways, not a blocker for this PR as it's more of a design question)

Reply:
Thank you for your suggestion, I update in the `gemma_full_finetune.py`.